index.js

```diff
@@ -0,0 +1,471 @@
+  //.
+  //. > A value which has an Applicative must provide an `of` method on
+  //. > itself or its `constructor` object.
+  var Applicative =
+  TypeClass('sanctuary-type-classes/Applicative',
+            [Apply],
+            either(hasMethod(FL.of), hasStaticMethod(FL.of)));
+
+  //# Chain :: TypeClass
+  //.
+  //. > A value that implements the Chain specification must also implement
+  //. > the Apply specification.
+  //.
+  //. > A value which has a Chain must provide a `chain` method.
+  var Chain =
+  TypeClass('sanctuary-type-classes/Chain',
+            [Apply],
+            hasMethod(FL.chain));
+
+  //# Monad :: TypeClass
+  //.
+  //. > A value that implements the Monad specification must also implement
+  //. > the Applicative and Chain specifications.
+  var Monad =
+  TypeClass('sanctuary-type-classes/Monad',
+            [Applicative, Chain],
+            K(true));
+
+  //# Extend :: TypeClass
+  //.
+  //. > An Extend must provide an `extend` method.
+  var Extend =
+  TypeClass('sanctuary-type-classes/Extend',
+            [],
+            hasMethod(FL.extend));
+
+  //# Comonad :: TypeClass
+  //.
+  //. > A value that implements the Comonad specification must also implement
+  //. > the Functor and Extend specifications.
+  //.
+  //. > A value which has a Comonad must provide an `extract` method on itself.
+  var Comonad =
+  TypeClass('sanctuary-type-classes/Comonad',
+            [Functor, Extend],
+            hasMethod(FL.extract));
+
+  //. ### Built-in types
+  //.
+  //. This package provides implementations of each method for each built-in
+  //. type which can satisfy the requirements of the corresponding type class.
+  //.
+  //. Function can satisfy the requirements of Functor, for example, so this
+  //. package implements a `map` method for Function, to which the exported
+  //. `map` function dispatches when appropriate.
+  //.
+  //. | Type class      |   Array   | Function  |  Object   |  String   |
+  //. |:--------------- |:---------:|:---------:|:---------:|:---------:|
+  //. | [Setoid][]      |           |           |           |           |
+  //. | [Semigroup][]   |     ✔︎     |           |     ✔︎     |     ✔︎     |
+  //. | [Monoid][]      |     ✔︎     |           |     ✔︎     |     ✔︎     |
+  //. | [Functor][]     |     ✔︎     |     ✔︎     |     ✔︎     |           |
+  //. | [Foldable][]    |     ✔︎     |           |           |           |
+  //. | [Traversable][] |     ✔︎     |           |           |           |
+  //. | [Apply][]       |     ✔︎     |     ✔︎     |           |           |
+  //. | [Applicative][] |           |           |           |           |
+  //. | [Chain][]       |     ✔︎     |           |           |           |
+  //. | [Monad][]       |           |           |           |           |
+  //. | [Extend][]      |           |           |           |           |
+  //. | [Comonad][]     |           |           |           |           |
```

Comment:
| Type class      |   Array   | Function  |  Object   |  String   |
|:--------------- |:---------:|:---------:|:---------:|:---------:|
| [Setoid][]      |           |           |           |           |
| [Semigroup][]   |     ✔︎     |           |     ✔︎     |     ✔︎     |
| [Monoid][]      |     ✔︎     |           |     ✔︎     |     ✔︎     |
| [Functor][]     |     ✔︎     |     ✔︎     |     ✔︎     |           |
| [Foldable][]    |     ✔︎     |           |           |           |
| [Traversable][] |     ✔︎     |           |           |           |
| [Apply][]       |     ✔︎     |     ✔︎     |           |           |
| [Applicative][] |           |           |           |           |
| [Chain][]       |     ✔︎     |           |           |           |
| [Monad][]       |           |           |           |           |
| [Extend][]      |           |           |           |           |
| [Comonad][]     |           |           |           |           |

Could we check any more of these cells? I think I've implemented all the ones Ramda does, plus `Object#concat` (which is known as `merge` in Ramda).


[Applicative]:    https://github.com/fantasyland/fantasy-land#applicative
[Apply]:          https://github.com/fantasyland/fantasy-land#apply
[Chain]:          https://github.com/fantasyland/fantasy-land#chain
[Comonad]:        https://github.com/fantasyland/fantasy-land#comonad
[Extend]:         https://github.com/fantasyland/fantasy-land#extend
[Foldable]:       https://github.com/fantasyland/fantasy-land#foldable
[Functor]:        https://github.com/fantasyland/fantasy-land#functor
[Monad]:          https://github.com/fantasyland/fantasy-land#monad
[Monoid]:         https://github.com/fantasyland/fantasy-land#monoid
[Semigroup]:      https://github.com/fantasyland/fantasy-land#semigroup
[Setoid]:         https://github.com/fantasyland/fantasy-land#setoid
[Traversable]:    https://github.com/fantasyland/fantasy-land#traversable